be/src/olap/task/engine_clone_task.cpp

```diff
@@ -847,10 +844,19 @@ OLAPStatus EngineCloneTask::_clone_full_data(TabletSharedPtr tablet, TabletMeta*
     // in previous step, copy all files from CLONE_DIR to tablet dir
     // but some rowset is useless, so that remove them here
     for (auto& rs_meta_ptr : rs_metas_found_in_src) {
-        RowsetSharedPtr org_rowset(new AlphaRowset(&(cloned_tablet_meta->tablet_schema()), 
-            tablet->tablet_path(), tablet->data_dir(), rs_meta_ptr));
-        if (org_rowset->init() == OLAP_SUCCESS && org_rowset->load() == OLAP_SUCCESS) {
-            org_rowset->remove();
+        RowsetSharedPtr rowset_to_remove;
+        auto s = RowsetFactory::create_rowset(&(cloned_tablet_meta->tablet_schema()),
+                                              tablet->tablet_path(),
+                                              tablet->data_dir(),
+                                              rs_meta_ptr,
+                                              &rowset_to_remove);
+        if (s != OLAP_SUCCESS) {
+            LOG(WARNING) << "failed to init rowset to remove: " << rs_meta_ptr->rowset_id().to_string();
+            continue;
+        }
```

Comment:
forget load?

Reply:
Nope. There is no need to call load() before remove().